### PR TITLE
Create or update registration contact on Vumi

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: control/settings.py [hunk 1]
```diff
@@ -292,6 +293,12 @@ def abspath(*args):
     },
     'subsend.tasks.processes_message': {
         'queue': 'lowpriority',
+    },
+    'registration.tasks.jembi_post_json': {
+        'queue': 'lowpriority',
```
Comment (imsickofmaps) [inline control/settings.py]: these should actually be `highpriority`

File: registration/models.py [hunk 2]
```diff
@@ -75,6 +75,9 @@ def clean(self):
         if self.mom_id_type == 'sa_id' and self.mom_id_no is None:
             raise ValidationError(
                 _("Provide an id number in the mom_id_no field."))
+        if self.mom_id_type == 'sa_id' and self.mom_dob is None:
```
Comment (imsickofmaps) [inline registration/models.py:78]: Is this constraint needed, given that we can extract the DOB from the ID?

File: registration/tests.py [hunk 3]
```diff
@@ -7,12 +7,24 @@
...
 
 
-Jembi_Post_Json.get_timestamp = lambda x: "20130819144811"
+def override_get_timestamp():
+    return "20130819144811"
+
+tasks.get_timestamp = override_get_timestamp
+
+
+def override_get_tomorrow():
```
Comment (imsickofmaps) [inline registration/tests.py:22]: Please make the two timestamp overrides logically aligned

File: registration/tests.py [hunk 4]
```diff
@@ -369,7 +499,7 @@ def test_build_jembi_json_clinic_self(self):
             'type': 3,
             'swt': 1
         }
-        json = jembi_post_json.build_jembi_json(reg)
+        json = tasks.build_jembi_json(reg)
```
Comment (imsickofmaps) [inline registration/tests.py]: careful using core language module names as variable names - better to use something unambiguous

File: registration/tests.py [hunk 5]
```diff
@@ -459,6 +589,112 @@ def test_jembi_post_json(self):
...
+            u"key": u"knownuuid",
+            u"msisdn": u"+27001",
+            u"user_account": u"knownaccount",
+            u"extra": {}
+        })
+
+        contact = tasks.update_create_vumi_contact.apply_async(
+            kwargs={"registration_id": registration.data["id"],
+                    "client": client})
+        self.assertEqual(contact.get()["msisdn"], "+27001")
```
Comment (imsickofmaps) [inline registration/tests.py]: This works of course, but you could simplify this for yourself by doing `result = contact.get()` and then just using `result["key"], "expected"` throughout.

**Comment by imsickofmaps** [comment]
first pass done, some small fixes, then I'll re-review

File: registration/tests.py (hunk 4 above)
Comment (gsvr) [inline registration/tests.py]: Good point!

File: registration/models.py (hunk 2 above)
Comment (gsvr) [inline registration/models.py:78]: Opened issue #97

**Comment by gsvr** [comment]
@imsickofmaps Ready for re-review.

File: registration/tests.py [hunk 6]
```diff
@@ -7,12 +7,23 @@
...
-Jembi_Post_Json.get_timestamp = lambda x: "20130819144811"
+def override_get_timestamp():
+    return "20130819144811"
+
+
+def override_get_tomorrow():
+    return "2014-01-02"
+
+tasks.get_timestamp = override_get_timestamp
+tasks.get_tomorrow = override_get_tomorrow
```
Comment (imsickofmaps) [inline registration/tests.py:26]: ROFL! I didn't mean the file alignment, I meant the fact that one day is 2014 and one is 2013 so it's not really "yesterday"!

**Comment by imsickofmaps** [comment]
One minor - then :+1:

File: registration/tests.py (hunk 6 above)
Comment (gsvr) [inline registration/tests.py:26]: @imsickofmaps Oh! Haha :laughing: I hear what you're saying, however the different dates are used for calibration in the next issue (#90) - I'm repeating the same tests that we had in JS code to validate that we're getting the same results.  So I'm going to land this here and if you're happy I'm getting the correct results in the next issue I can change it then.

Comment (imsickofmaps) [inline registration/tests.py:26]: OK

**Pull request merged**